source/geod24/concurrency.d

```diff
@@ -985,6 +986,161 @@ protected:
         bool notified;
     }
 
+    class Channel (T)
+    {
+        this(ulong size = 0)
+        {
+            this.size = size;
+            m_lock = new Mutex;
+        }
+
+        bool write(T val)
+        {
+            bool notify_peer, fiber_was_blocked = false;
```

Comment:
By the way, this only sets `fiber_was_blocked` to `false` explicitly. It leaves `notify_peer` to equal `bool.init`. But `bool.init` is already false. I suggest moving each variable declaration to its own line (it follows our style guide), and we can avoid initializers when `Type.init` is already what we want.

```suggestion
            bool notify_peer;
```

Also the `fiber_was_blocked` declaration should be moved to where it's first used. This is also our coding style: we do not declare variables at the start of the function, but instead to the closest where they are initialized.